internal/tracinginterceptor/interceptor.go

```diff
@@ -177,12 +177,55 @@ func (i *Interceptor) CallOneway(ctx context.Context, req *transport.Request, ou
 
 // HandleStream implements interceptor.StreamInbound
 func (i *Interceptor) HandleStream(s *transport.ServerStream, h transport.StreamHandler) error {
-	panic("implement me")
+	parentSpanCtx, _ := i.tracer.Extract(i.propagationFormat, getPropagationCarrier(s.Request().Meta.Headers.Items(), i.transport))
+	span := i.tracer.StartSpan(
+		s.Request().Meta.Procedure,
+		opentracing.ChildOf(parentSpanCtx),
+		ext.SpanKindRPCServer,
+		opentracing.StartTime(time.Now()),
+	)
+	defer span.Finish()
+	err := h.HandleStream(s)
+
+	isApplicationError := err != nil
+	var appErrorMeta *transport.ApplicationErrorMeta
+	if isApplicationError {
+		code := yarpcerrors.FromError(err).Code()
+		appErrorMeta = &transport.ApplicationErrorMeta{
+			Code:    &code,
+			Name:    s.Request().Meta.Procedure,
+			Details: err.Error(),
+		}
+	}
+
+	return updateSpanWithErrorDetails(span, isApplicationError, appErrorMeta, err)
 }
 
 // CallStream implements interceptor.StreamOutbound
 func (i *Interceptor) CallStream(ctx context.Context, req *transport.StreamRequest, out transport.StreamOutbound) (*transport.ClientStream, error) {
-	panic("implement me")
+	span := i.tracer.StartSpan(
+		req.Meta.Procedure,
+		ext.SpanKindRPCClient,
+		opentracing.StartTime(time.Now()),
+	)
+	defer span.Finish()
+	tracingHeaders := make(map[string]string)
+	if err := i.tracer.Inject(span.Context(), i.propagationFormat, getPropagationCarrier(tracingHeaders, i.transport)); err != nil {
+		span.LogFields(logFieldEventError, log.String("message", err.Error()))
+	} else {
+		for k, v := range tracingHeaders {
+			req.Meta.Headers = req.Meta.Headers.With(k, v)
+		}
+	}
+
+	clientStream, err := out.CallStream(ctx, req)
```

Comment:
We need `tracedClientStream` wrapper here. The current code only captures the client stream creation error, but not the error from the actual communication - `SendMsg` or `RecvMsg`.

Reply:
+1. commented on slack thread, not quite sure how to handle the HandleStream though. Since h.HandleStream accepts *ServerStream while tracedStream is of type *tracedServerStream. Same question for client side.

Reply:
This is how you can achieve for inbound: [observability middleware example](https://github.com/yarpc/yarpc-go/blob/31a6e88b74a5211b9a0f22b594d4e8fd7b50c801/internal/observability/stream.go#L69), but I don't see the necessity of server stream wrapper for inbound.